core/src/main/java/tech/pegasys/web3signer/core/multikey/metadata/AbstractArtifactSignerFactory.java

```diff
@@ -53,12 +55,21 @@ protected AbstractArtifactSignerFactory(
   }
 
   protected Bytes extractBytesFromVault(final AzureSecretSigningMetadata metadata) {
-    final AzureKeyVault azureVault =
-        new AzureKeyVault(
-            metadata.getClientId(),
-            metadata.getClientSecret(),
-            metadata.getTenantId(),
-            metadata.getVaultName());
+    final AzureKeyVault azureVault;
+    if (metadata.getAuthenticationMode() == AzureAuthenticationMode.CLIENT_SECRET) {
+      azureVault =
+          AzureKeyVault.createUsingClientSecretCredentials(
+              metadata.getClientId(),
+              metadata.getClientSecret(),
+              metadata.getTenantId(),
+              metadata.getVaultName());
+    } else {
+      final Optional<String> clientId =
```

Comment:
this could be removed if the AzureSecretSigningMetadata clientId was optional field in the Config object